src/huggingface_hub/utils/_fixes.py

```diff
@@ -40,7 +40,7 @@ def SoftTemporaryDirectory(
     prefix: Optional[str] = None,
     dir: Optional[Union[Path, str]] = None,
     **kwargs,
-) -> Generator[str, None, None]:
+) -> Generator[Path, None, None]:
```

Comment:
It's usually dangerous to update the return value of an helper but in this case it looks fine :+1:

(+1 on using Path as much as possible. Codebase is currently a mix of str and Path for legacy reasons but 🤷)